### PR TITLE
Fix S2077 FN: Add support for Mono.Data.Sqlite

### LINKED ISSUE #7261: Fix S2077 FN: Add support for Mono.Data.Sqlite
### Description

Hey dotnet artists!

While tackling [APPSEC-725](https://sonarsource.atlassian.net/browse/APPSEC-725) for the WebGoat framework, it appeared that we did not support the following two method IDs in this rule:

``` C#
Mono.Data.Sqlite.SqliteDataAdapter.SqliteDataAdapter(string, Mono.Data.Sqlite.SqliteConnection)
Mono.Data.Sqlite.SqliteCommand.SqliteCommand(string, Mono.Data.Sqlite.SqliteConnection)
```

Cheers!

### Repro steps

This appears in the [WebGoat.Net](https://github.com/jerryhoff/WebGoat.NET/tree/1c6cab19f9029673cd98ba8624bf9cc91d04bae9) project, in this file: https://github.com/jerryhoff/WebGoat.NET/blob/1c6cab19f9029673cd98ba8624bf9cc91d04bae9/WebGoat/App_Code/DB/SqliteDbProvider.cs.

Here is a reproducer:

``` C#
using Mono.Data.Sqlite;

namespace Reproducer
{
    public class S2077
    {
        void Reproduce(string connectionString, string query)
        {
            //get data
            string sql = "select * from table where query = '" + query + "';";
            
            using (SqliteConnection connection = new SqliteConnection(connectionString))
            {
                connection.Open();

                SqliteDataAdapter da = new SqliteDataAdapter(sql, connection); // FN: S2077

                SqliteCommand command = new SqliteCommand(sql, connection); // FN: S2077
            }              
        }
    }
}
```

### Related information

* .Net Framework v3.5

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes #7261

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Linq;
+using Mono.Data.Sqlite;
+
+namespace Test
+{
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs:6]: Remove the namespace, kill the noise.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public class C
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs:7]: Maybe rename the class to `Sample` for consistency with other tests. `C` looks pretty arbitrary and confusing.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            var command = new SqliteCommand();       // Compliant
+            command = new SqliteCommand(connection); // Compliant
+            var adapter = new SqliteDataAdapter();   // Compliant
+        }
+
+        void Foo(SqliteConnection connection, string query, params object[] parameters)
+        {
+            var command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection);  // Noncompliant
+            command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}");                  // Noncompliant
+            var adapter = new SqliteDataAdapter(string.Concat(query, parameters), connection);          // Noncompliant
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs:22]: Add a test for the overload taking `(string, string)`.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            var command = new SqliteCommand();       // Compliant
+            command = new SqliteCommand(connection); // Compliant
+            var adapter = new SqliteDataAdapter();   // Compliant
+        }
+
+        void Foo(SqliteConnection connection, string query, params object[] parameters)
+        {
+            var command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection);  // Noncompliant
+            command = new SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}");                  // Noncompliant
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.cs:21]: Add a UT for the overload taking `(string, SQLiteTransaction)`.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.vb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        Private Sub Compliant(ByVal connection As SqliteConnection)
+            Dim command = New SqliteCommand()       ' Compliant
+            command = New SqliteCommand(connection) ' Compliant
+            Dim adapter = New SqliteDataAdapter()   ' Compliant
+        End Sub
+
+        Private Sub Foo(ByVal connection As SqliteConnection, ByVal query As String, ParamArray parameters As Object())
+            Dim command = New SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}", connection) ' Noncompliant
+            command = New SqliteCommand($"SELECT * FROM mytable WHERE mycol={query}")                 ' Noncompliant
+            Dim adapter = New SqliteDataAdapter(String.Concat(query, parameters), connection)         ' Noncompliant
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.vb:18]: Add tests for the overloads mentioned in the C# UT.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.vb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿Imports System
+Imports System.Linq
+Imports Mono.Data.Sqlite
+
+Namespace Test
+    Public Class C
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.MonoSqlLite.vb:6]: Remove the namespace, rename the class. Same as for C#.

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
There are a few overloads that still need to be covered.

**Review by Tim-Pohlmann — APPROVED** [review]
I just realized you should also add compliant test cases for unformatted strings (literal or variable).
Everything else LGTM.

**Comment by martin-strecker-sonarsource** [comment]
> I just realized you should also add compliant test cases for unformatted strings (literal or variable).

There is https://github.com/SonarSource/sonar-dotnet/blob/2197ff3c45160f42ce0656e27a2b125a558c05a5/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.Net46.cs which covers all these cases. The code paths are the same for Mono.SqlLite. I don't see the value to repeat all the cases again.

**Pull request merged**